subsys/ipc/ipc_service/backends/ipc_rpmsg_static_vrings.c

```diff
@@ -448,10 +469,86 @@ static int open(const struct device *instance)
 
 }
 
+static int get_tx_buffer(const struct device *instance, void *token,
+			 void **data, uint32_t *len, k_timeout_t wait)
+{
+	struct ipc_rpmsg_ept *rpmsg_ept;
+	uint32_t max_size;
+	void *payload;
+	int bin_wait;
+
+	if (!data || !len) {
+		return -EINVAL;
+	}
+
+	/* OpenAMP only supports a binary wait / no-wait */
+	if (!K_TIMEOUT_EQ(wait, K_FOREVER) && !K_TIMEOUT_EQ(wait, K_NO_WAIT)) {
+		return -EINVAL;
```

Comment:
`-EINVAL` should be returned only for invalid parameter values, and according to API description any timeout value is valid.
I think this function should implement some kind of retry mechanism if timeout has other value then FOREVER of NO_WAIT.

Reply:
I really wanted to avoid that TBH because I think that will open a can of worms. What about adding a new error instead for unsupported timeout?

Reply:
I think `-ENOTSUP` or something similar would be fine. Retry could be implemented later